dask-gateway-server/dask-gateway-proxy/tools/fail_without_go_1_12.go

```diff
@@ -1,3 +1,4 @@
+//go:build !go1.12
 // +build !go1.12
```

Comment:
It would be perfectly fine to require golang 1.17+ for example. I think overall, declaring that we require 1.12 but not testing against 1.12 can just be a reason for possible issues.

Currently I know we test compile the binary with 1.16, 1.17, and 1.18, and know from https://endoflife.date/go that only 1.17+ hasn't reached end of life.

https://github.com/dask/dask-gateway/blob/01eef766f05d3ff91718b134ccb9490b02f50671/.github/workflows/test.yaml#L58-L63

Reply:
I currently don't have a good enough understanding of the purpose of `fail_without_go_1_12.go` or where/how it's intended to be used, which is why I relocated it to the `tools/` directory.

Reply:
Ah, neither do i. Perhaps it was a known requirement to have 1.12 at some point, and meant to provide a helpful failure message?

Good to know you dont recognize this practice, i figure then it makes most sense to just remove it at this point then.

Reply:
Doing that is feature creep, this is good to go as it is i think as well.